libexec/python/defaults.py

```diff
@@ -50,7 +50,8 @@ def getenv(variable_key, default=None, required=False, silent=False):
         sys.exit(1)
 
     if silent:
-        bot.verbose2("%s found" % (variable_key))
+        if variable is not None:
+            bot.verbose2("%s found" % (variable_key))
     else:
```

Comment:
I was first worried we would be hiding good debugging into (e.g., I would want to know when a variable is set to None) but it probably communicates that by way of not printing, so I like this approach better to produce cleaner python logs. I'm down to give it a try! The variable silent is poorly named (my bad-doing) because it's intended to hide sensitive information like passwords, tokens, while still alerting the user that it's set. This is different than performing the print action - because the message bot inherits the `SINGULARITY_MESSAGELEVEL`.

Reply:
Thanks for explaining the meaning of `silent`, I didn't know! That said, this change is independent of that, it simply says not to print `VARIABLE found` unless the variable was in fact found (`variable is not None`, which I guess I could simplify to `if variable:`). Before, it would print `VARIABLE found` for all variables even when **not** found, which kind of defeats the purpose. I assume it was a small typo, and I needed it fixed to debug my changes.
Notice it only prints the name of the variable, not its value, so there shouldn't be any password or token leak (and I didn't write that code, I only added the if condition).

Reply:
yes this is also good! I definitely like it much better - the log will be a lot cleaner, the user doesn't actually learn much to see a variable set to None.

Reply:
While that is true, I think you're missing the point. What I was seeing was `SOME_VARIABLE found` when `SOME_VARIABLE` was not found at all (but not required, else it would exit)! That's not just unnecessary noise in the log, that's actually misleading.

Reply:
Agree. I meant to say to python, it's the same. To the user, I agree it's misleading and this is a good change.